enhancements/console/samples.md

```diff
@@ -0,0 +1,396 @@
+
+### API Extensions
+
+The console operator automatically installs a new `ConsoleSample` CRD.
+
+### Risks and Mitigations
+
+#### The OpenShift Console is optional and could be disabled
+
+Other operators should expect that the `ConsoleSample` CRD is unavailable like the other `Console*` CRDs.
+
+### Drawbacks
+
+Another extra (but specific) custom resource just for console samples.
+
+## Design Details
+
+An "Import from Git" example of how we have it today as part of the Builder Images annotation:
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: java-sample
+spec:
+  title: Java
+  abstract: Build and run Java applications using Maven and OpenJDK.
+  description: |
+    # About this sample
+    Build and run Java applications using Maven and OpenJDK.
+    # Links
+    * [Sample repository](https://github.com/jboss-openshift/openshift-quickstarts)
+  icon: base64 encoded image
+  provider: Red Hat
+  badge: Source to Image
+  tags:
+  - java
+  - jboss
+  - openjdk
+  source:
+    type: GitImport
+    gitImport:
+      url: https://github.com/jboss-openshift/openshift-quickstarts
+```
+
+An "Import container image" example:
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: minimal-ubi-container
+spec:
+  title: Minimal UBI container
+  abstract: Test the minimal Red Hat Universal Base Image (UBI).
+  description: |
+    # About this sample
+    The Red Hat Universal Base Image is free to deploy on Red Hat or non-Red Hat platforms
+    and freely redistributable...
+  icon: base64 encoded image
+  provider: Red Hat
+  badge: UBI Container
+  source:
+    type: ContainerImport
+    containerImport:
+      image: registry.access.redhat.com/ubi8/ubi-minimal:8.8-860
+```
+
+### Attributes
+
+The following constraints are checked in the [openshift/api change](https://github.com/openshift/api/pull/1503) as well:
+
+1. `title` is the display name of the sample. It is required and can have a maximum length of **50** characters.
+2. `abstract` is a one-liner that describes the sample.
+   It is required and is shown on the sample card tile below the title and provider.
+3. `description` is required, has no maximum length, and is a README.md-like content for
+   additional information, links, pre and post conditions and instrucations.
+   It will be rendered as **Markdown** so that it can contain line breaks, links, and other simple formatting.
+4. `icon` is optional and must be a base64 encoded image.
+
+   The icon will be displayed beside the sample title and should be a vector image for easy scaling.
+   It will be shown on a white background (also when the dark theme is used)
+   and in different sizes (currently not bigger than 40x100px).
+   Most icons will be quadratic, but the web console ensures that all aspect ratios work.
+
+5. `provider` like "Red Hat" is optional and can have a maximum length of **20** characters.
```

Comment:
This doesn't really explain what a Provider is, how would you describe the provider?

Reply:
Does this help?

6. `provider` is an optional label to honor who provides the sample. It can have a maximum length of **20** characters.

   A provider can be a company like "Red Hat" or an organization like "CNCF" or "Knative".

   Currently, the provider is only shown on the sample card tile below the title with the prefix "Provided by " (see screenshot above).

Reply:
Yeah that's much better! Thanks